semantic-python/src/Language/Python/Core.hs

```diff
@@ -90,30 +93,13 @@ locate syn item = do
 
   pure (Core.annAt (locFromTSSpan (getField @"ann" syn)) item)
 
--- | TODO: This is not right, it should be a reference to a Preluded
--- NoneType instance, but it will do for now.
-none :: (Member Core sig, Carrier sig t) => t Name
-none = unit
-
--- | Helper for delegating to compileCC. The presence of this function indicates
--- that we might want to move 'compile' out of the Compile class entirely.
-viaCompileCC :: ( Compile py
-                , CoreSyntax syn t
-                , Member (Reader SourcePath) sig
-                , Member (Reader Bindings) sig
-                , Carrier sig m
-                , MonadFail m
-                )
-             => py -> m (t Name)
-viaCompileCC t = compileCC t (pure none)
-
 defaultCompile :: (MonadFail m, Show py) => py -> m (t Name)
 defaultCompile t = fail $ "compilation unimplemented for " <> show t
 
 newtype CompileSum py = CompileSum py
 
 instance (Generic py, GCompileSum (Rep py)) => Compile (CompileSum py) where
-  compile (CompileSum a) = gcompileSum . from $ a
+  compileCC (CompileSum a) cc = gcompileCCSum (from a) cc
   compileCC (CompileSum a) cc = gcompileCCSum (from a) cc
```

Comment:
The identical definitions suggest that a) we can delete one of them and b) we probably need more warnings enabled.

Reply:
I don’t think we’re compiling with `-Werror` on CI. Should fix that.